### ISSUE TITLE
Void type in some scenarios crashes the compiler

### ISSUE THREAD

**Issue by romdotdog** [AssemblyScript/assemblyscript#2208]
## Credits

@DogWhich also found this interesting one on a whim

## Code

```ts
typeof ((): void => {})();
```

[Try it in the playground](https://www.assemblyscript.org/editor.html#IyFvcHRpbWl6ZT1zcGVlZCZydW50aW1lPXN0dWIKdHlwZW9mICgoKTogdm9pZCA9PiB7fSkoKTsKIyFodG1sCjx0ZXh0YXJlYSBpZD0ib3V0cHV0IiBzdHlsZT0iaGVpZ2h0OiAxMDAlOyB3aWR0aDogMTAwJSIgcmVhZG9ubHk+PC90ZXh0YXJlYT4KPHNjcmlwdD4KbG9hZGVyLmluc3RhbnRpYXRlKG1vZHVsZV93YXNtLCB7IC8qIGltcG9ydHMgKi8gfSkKICAudGhlbigoeyBleHBvcnRzIH0pID0+IHsKICAgIGNvbnN0IG91dHB1dCA9IGRvY3VtZW50LmdldEVsZW1lbnRCeUlkKCdvdXRwdXQnKQogICAgZm9yIChsZXQgaSA9IDA7IGkgPD0gMTA7ICsraSkgewogICAgICBvdXRwdXQudmFsdWUgKz0gYGZpYigke2l9KSA9ICR7ZXhwb3J0cy5maWIoaSl9XG5gCiAgICB9CiAgfSkKPC9zY3JpcHQ+)

## Stack trace

```
▌ Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ Here is the stack trace hinting at the problem, perhaps it's useful?
▌ 
▌ AssertionError: assertion failed
▌     at assert (D:\assemblyscript\std\portable\index.js:200:11)
▌     at Compiler.convertExpression (D:\assemblyscript\src\compiler.ts:3598:7)
▌     at Compiler.compileTypeof (D:\assemblyscript\src\compiler.ts:9948:25)
▌     at Compiler.compileUnaryPrefixExpression (D:\assemblyscript\src\compiler.ts:9889:21)
▌     at Compiler.compileExpression (D:\assemblyscript\src\compiler.ts:3539:21)
▌     at Compiler.compileExpressionStatement (D:\assemblyscript\src\compiler.ts:2521:17)
▌     at Compiler.compileStatement (D:\assemblyscript\src\compiler.ts:2213:21)
▌     at Compiler.compileTopLevelStatement (D:\assemblyscript\src\compiler.ts:2175:25)
▌     at Compiler.compileFile (D:\assemblyscript\src\compiler.ts:1111:12)
▌     at Compiler.compile (D:\assemblyscript\src\compiler.ts:486:14)
▌     at Object.compile (D:\assemblyscript\src\index.ts:280:32)
▌     at D:\assemblyscript\cli\asc.js:902:31
▌     at measure (D:\assemblyscript\cli\asc.js:1409:3)
▌     at Object.main (D:\assemblyscript\cli\asc.js:900:24)
▌     at runTest (D:\assemblyscript\tests\compiler.js:178:7)
▌ 
▌ If it refers to the dist files, try to 'npm install source-map-support' and
▌ run again, which should then show the actual code location in the sources.
▌ 
▌ If you see where the error is, feel free to send us a pull request. If not,
▌ please let us know: https://github.com/AssemblyScript/assemblyscript/issues
▌ 
▌ Thank you!
```

**Comment by romdotdog** [comment]
Here's another test-case:
```ts
((): void => {})() ? 0 : 0
```

**Comment by romdotdog** [comment]
Yet another crash:
```ts
store<void>(1, ((): void => {})());
```

**Comment by MaxGraey** [comment]
Original issue `typeof ((): void => {})();` was fixed after #2412

Next ones still need to fix:
```ts
((): void => {})() ? 0 : 0
store<void>(1, ((): void => {})());
```
